projects/plugins/jetpack/modules/subscriptions/views.php

```diff
@@ -246,14 +238,24 @@ static function render_widget_status_messages( $instance ) {
 			}
 
 			$border_color = isset( $themecolors['border'] ) ? " #{$themecolors['border']}" : '';
+
+			$redirect_fragment = self::get_wpcom_redirect_fragment();
 			printf(
-				'<div class="jetpack-sub-notification" style="border: 1px solid%1$s; padding-left: 5px; padding-right: 5px; margin-bottom: 10px;">%2$s</div>',
+				'<div id="%1$s" class="jetpack-sub-notification" style="border: 1px solid%2$s; padding-left: 5px; padding-right: 5px; margin-bottom: 10px;">%3$s</div>',
+				esc_attr( $redirect_fragment ),
 				esc_attr( $border_color ),
 				wp_kses_post( $message )
 			);
 		}
 	}
 
+	/**
+	 * Generates the redirect fragment used after form submission.
+	 */
+	protected static function get_wpcom_redirect_fragment() {
```

Comment:
We looked at this together today; @andres-blanco should be able to do an update that has no WP.com references, works the same on both JP and WP.com, and doesn't require the script at all in the PR.